gems/pending/spec/appliance_console/logical_volume_management_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'appliance_console/logical_volume_management'
+require 'pathname'
+require 'tmpdir'
+
+describe ApplianceConsole::LogicalVolumeManagement do
+  before do
+    @spec_name = File.basename(__FILE__).split(".rb").first
+    @disk_double = double(@spec_name, :path => "/dev/vtest")
+    @config = described_class.new(:disk => @disk_double, :mount_point => "/mount_point", :name => "test")
+  end
+
+  describe ".new" do
+    it "ensures required disk option is provided" do
+      expect { described_class.new(:mount_point => "/mount_point", :name => "test") }.to raise_error(ArgumentError)
+    end
+
+    it "ensures required mount_point option is provided" do
+      expect { described_class.new(:disk => @disk_double, :name => "test") }.to raise_error(ArgumentError)
+    end
+
+    it "ensures required name option is provided" do
+      expect do
+        described_class.new(:disk => @disk_double, :mount_point => "/mount_point")
+      end.to raise_error(ArgumentError)
+    end
+
+    it "sets derived and default instance variables automatically" do
+      expect(@config.volume_group_name).to eq("vg_test")
+      expect(@config.filesystem_type).to eq("xfs")
+      expect(@config.logical_volume_path).to eq("/dev/vg_test/lv_test")
+    end
+  end
+
+  describe "#setup" do
+    before do
+      expect(@disk_double).to receive(:create_partition_table)
+      expect(@disk_double).to receive(:partitions).and_return([:fake_partition])
+      @config.disk = @disk_double
+
+      @fstab = double(@spec_name)
+      allow(@fstab).to receive_messages(:entries => [])
+      allow(LinuxAdmin::FSTab).to receive_messages(:instance => @fstab)
+
+      expect(AwesomeSpawn).to receive(:run!).with("parted -s /dev/vtest mkpart primary 0% 100%")
```

Comment:
I'm not a fan of having expectations in the `before` block, but I wouldn't block the PR just for that.